lib/mongoid/clients/sessions.rb

```diff
@@ -41,12 +41,20 @@ module Sessions
       #
       # @since 6.4.0
       def with_session(options = {})
-        raise Mongoid::Errors::InvalidSessionUse.new(:invalid_session_nesting) if Threaded.get_session
+        if Threaded.get_session
+          raise Mongoid::Errors::InvalidSessionUse.new(:invalid_session_nesting)
+        end
         session = persistence_context.client.start_session(options)
         Threaded.set_session(session)
         yield(session)
       rescue Mongo::Error::InvalidSession => ex
-        if ex.message == Mongo::Session::SESSIONS_NOT_SUPPORTED
+        if
+          # Driver 2.13.0+
+          defined?(Mongo::Error::SessionsNotSupported) &&
+            Mongo::Error::SessionsNotSupported === ex or
```

Comment:
Is there a reason to use `or` here vs `||`? All things being equal, I think I'd prefer consistent styles.

Reply:
I think I did it for emphasis but I can redo with punctuation.

Reply:
Thanks!